bundles/binding/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/LoginData.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2010-2015, openHAB.org and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.myq.internal;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This Class parses the JSON data and stores if the login request was
+ * successful.
+ * <ul>
+ * <li>success: json request was successful</li>
+ * <li>securityToken: securityToken from login request</li>
+ * </ul>
+ * 
+ * @author Scott Hanson
+ * @since 1.8.0
+ */
+public class LoginData {
+	static final Logger logger = LoggerFactory.getLogger(LoginData.class);
+
+	String securityToken;
+
+	/**
+	 * Constructor of the LoginData.
+	 * 
+	 * @param loginData
+	 *            The Json string as it has been returned myq website.
+	 */
+	@SuppressWarnings("unchecked")
+	public LoginData(JsonNode root) throws IOException {
+		ObjectMapper mapper = new ObjectMapper();
+		Map<String, Object> treeData = mapper.readValue(root, Map.class);
+		Object data = treeData.get("SecurityToken");
+		if(data == null)
+			throw new IOException("Could not find SecurityToken in JSON data");
```

Comment:
I think it would improve readability to formally use an exception type that represents the class of exception this is, which could be about malformed JSON.  I worry that higher catch blocks could take an IOException at face value that there are network issues, when in fact the I/O is fine but the JSON was not as expected.  For example, if a Java library class unexpectedly threw an IOException when there way no I/O problem, a catch block could potentially tear down and reconnect a socket that was not the problem at all.  Can higher catch blocks catch any shared parent class like Exception?